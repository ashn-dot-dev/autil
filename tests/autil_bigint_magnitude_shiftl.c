#define AUTIL_IMPLEMENTATION
#include "../autil.h"

int
main(void)
{
    struct autil_bigint* x = autil_bigint_new_cstr("0xff");
    autil_bigint_dump(x);

    autil_bigint_magnitude_shiftl(x, 0);
    autil_bigint_dump(x);

    autil_bigint_magnitude_shiftl(x, 1);
    autil_bigint_dump(x);

    autil_bigint_magnitude_shiftl(x, 7);
    autil_bigint_dump(x);

    autil_bigint_magnitude_shiftl(x, 4);
    autil_bigint_dump(x);

    struct autil_bigint* const y = autil_bigint_new_cstr("-0b1");
    autil_bigint_dump(y);
    autil_bigint_magnitude_shiftl(y, 1);
    autil_bigint_dump(y);

    struct autil_bigint* const z = autil_bigint_new(AUTIL_BIGINT_ZERO);
    autil_bigint_magnitude_shiftl(z, 1);
    autil_bigint_dump(z);

    struct autil_bigint* const w = autil_bigint_new_cstr("0b1");
    autil_bigint_dump(w);
    autil_bigint_magnitude_shiftl(w, 1);
    autil_bigint_dump(w);

    autil_bigint_del(x);
    autil_bigint_del(y);
    autil_bigint_del(z);
    autil_bigint_del(w);
    return EXIT_SUCCESS;
}
