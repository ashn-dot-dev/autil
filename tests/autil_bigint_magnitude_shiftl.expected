SIGN: '+', COUNT: 1, LIMBS: [0xFF]
SIGN: '+', COUNT: 1, LIMBS: [0xFF]
SIGN: '+', COUNT: 2, LIMBS: [0xFE, 0x01]
SIGN: '+', COUNT: 2, LIMBS: [0x00, 0xFF]
SIGN: '+', COUNT: 3, LIMBS: [0x00, 0xF0, 0x0F]
SIGN: '-', COUNT: 1, LIMBS: [0x01]
SIGN: '-', COUNT: 1, LIMBS: [0x02]
SIGN: '0', COUNT: 0, LIMBS: []
SIGN: '+', COUNT: 1, LIMBS: [0x01]
SIGN: '+', COUNT: 1, LIMBS: [0x02]
